components/supervisor/pkg/ports/served-ports.go

```diff
@@ -73,7 +74,10 @@ func (p *PollingServedPortsObserver) Observe(ctx context.Context) (<-chan []Serv
 			case <-ticker.C:
 			}
 
-			var ports []ServedPort
+			var (
+				visited map[string]struct{}
```

Comment:
```suggestion
visited = make(map[string]struct{})
```

this way we don't have to check if its nil and create it later.